tests/generic_relations/tests.py

```diff
@@ -555,6 +555,23 @@ def test_unsaved_instance_on_generic_foreign_key(self):
         with self.assertRaises(IntegrityError):
             TaggedItem.objects.create(tag="shiny", content_object=quartz)
 
+    def test_cache_invalidation_for_content_type_id(self):
+        # Create a Vegetable and Mineral with the same id.
+        new_id = max(Vegetable.objects.order_by('-id')[0].id,
+                     Mineral.objects.order_by('-id')[0].id) + 1
+        broccoli = Vegetable.objects.create(id=new_id, name="Broccoli")
+        diamond = Mineral.objects.create(id=new_id, name="Diamond", hardness=7)
```

Comment:
The explicit ID will probably be ignored on some third-party backends (MS SQL), which would make this test fail, but I didn't find a suitable database feature flag to detect those backends.